src/libraries/LiquidityNegation.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+/// @notice Library for negating a uint256 value.
+/// Used for decreasing liquidity on a pool.
+/// @dev It is safe to negate in an unchecked block because liquidity is always postive (uint256)
+library LiquidityNegation {
+    function toNegativeInt256(uint256 liquidity) internal pure returns (int256 liquidityDelta) {
+        unchecked {
+            liquidityDelta = -int256(liquidity);
```

Comment:
it still needs to us SafeCast though - the `uint256` -> `int256` isnt safe. Only the negation is

Reply:
Honestly imo not worth having a whole library for an unchecked subtraction. Can just go back to how it was if you prefer (not unchecked) , I jumping to another function call will probably undo the benefit.

Reply:
a jump is pretty cheap I thought? I was just trying to get rid of the ugliness with inline unchecked